### PR TITLE
Support MFA in Snowflake connections

### PR DESCRIPTION
**Pull request by sfc-gh-cnivera**
Resolves https://github.com/Snowflake-Labs/semantic-model-generator/issues/93

Most of the diff here is regenerating `poetry.lock`!

The majority of customers and SEs use accounts where MFA is enabled. For some MFA setups, the current authentication approach does not trigger a push notification for MFA on the user's device, which leads them to eventually be locked out of their Snowflake accounts.

This PR adds support for the `username_password_mfa` authenticator type (see the official [docs](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-api#label-snowflake-connector-methods) for more details). A user can set this authenticator environment variable alongside the passcode from their MFA app of choice in order to authenticate.

## Testing

Use the semantic model generator app with an account that has MFA enabled (instructions to do this are [here](https://select.dev/docs/snowflake-developer-guide/snowflake-mfa)). 

First, do not set any `SNOWFLAKE_AUTHENTICATOR` as if you were using a standard username/password. For some MFA setups, this will automatically send a push notification to your MFA app, which you can then approve. After approval, ensure that you can use the app normally.

Also confirm that the `username_password_mfa` authentication type works properly. To test this, set `SNOWFLAKE_AUTHENTICATOR="username_password_mfa"`, and `SNOWFLAKE_MFA_PASSCODE="<passcode-from-auth-app>"`. Verify that you do not get a push notification but are still able to use the app normally.

### CODE REVIEW

File: admin_apps/requirements.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ sqlglot==23.17.0
 streamlit==1.36.0
 streamlit-extras==0.4.3
 streamlit-monaco==0.1.3
-snowflake-connector-python==3.11.0
\ No newline at end of file
+snowflake-connector-python[secure-local-storage]==3.11.0
```
Comment (sfc-gh-cnivera) [inline admin_apps/requirements.txt:8]: Need `secure-local-storage` to fully support [MFA token caching](https://docs.snowflake.com/en/user-guide/security-mfa#label-mfa-token-caching).

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.9"]
+        python-version: [ "3.10" ]
```
Comment (sfc-gh-yayin) [inline .github/workflows/lint.yaml:14]: ooc, why do we need whitespace here?

**Review by sfc-gh-yayin — APPROVED** [review]
Tested this locally, i'm able to get the duo notification.

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (sfc-gh-cnivera) [reply .github/workflows/lint.yaml:14]: ah I think this was just my formatter running on save, should be a noop (but I did intentionally change the python version to be consistently 3.10)

**Pull request merged**